resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -462,6 +463,21 @@ private[yarn] class YarnAllocator(
 
   private def getHostAddress(nodeReport: NodeReport): String = nodeReport.getNodeId.getHost
 
+  /**
+   * Checks if node is in DECOMMISSIONING STATE
+   *
+   * Visible for testing.
+   */
+  def isDecommissioningNode(node: NodeReport): Boolean = {
+    // In hadoop-2.7 there is no support for node state DECOMMISSIONING
+    // In Hadoop-2.8, hadoop3.1 and later version of spark there is a support
+    // to node state DECOMMISSIONING.
+    // Inorder to build the spark using hadoop2 and hadoop3, not
+    // using YarnNodeState for the node state DECOMMISSIONING here and
+    // instead comparing Node states on spark end
+    NodeState.getYarnNodeState(node.getNodeState) == NodeState.DECOMMISSIONING
```

Comment:
hmm... Is it really necessary to define `NodeState`?

Is
```
NodeState.getYarnNodeState(node.getNodeState) == NodeState.DECOMMISSIONING
```
equivalent to 

```
node.getNodeState.toString.equals("DECOMMISSIONING")
```
?

Reply:
It seems that just change line 454 from 

```
node.getNodeState == NodeState.DECOMMISSIONING
```

to

```
node.getNodeState.toString.equals("DECOMMISSIONING")
``` 

and add some comments is enough